File: MainApp/src/app/core/models/UserModel.ts
import { DesignationModel } from './designation-model';

export interface UserModel {
  firstName: string;
  lastName: string;
  userName: string;
  email: string;
  imageUrl: string;
  designation: DesignationModel;
  lastMessage?: string;
  lastMessageTimeStamp?: Date;
  unreadMessageCount?: number;
  status: {
    id: number,
    type: string,
    tagClasses: string,
    tagStyle: string
  }
}

File: Models/UserModel.cs
using ChatApp.Business.Helpers;
using ChatApp.Context.EntityClasses;
using System;

namespace ChatApp.Models
{
    public class UserModel
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string UserName { get; set; }   

        public string Email { get; set; }

        public string ImageUrl { get; set; }
        public DesignationEntity Designation { get; set; }

        public string? LastMessage { get; set; } = null;

        public DateTime? LastMessageTimeStamp { get; set; } = null;

        public int? UnreadMessageCount { get; set; }

        public Status Status { get; set; }
    }
}
